supervisor/shared/serial.c

```diff
@@ -56,10 +56,15 @@ void serial_write_substring(const char* text, uint32_t length) {
     int errcode;
     common_hal_terminalio_terminal_write(&supervisor_terminal, (const uint8_t*) text, length, &errcode);
 #endif
-
+// split into chunks to avoid buffer overflows
     uint32_t count = 0;
+    uint32_t chunk = 0;
     while (count < length && tud_cdc_connected()) {
```

Comment:
should be splitted using the `tud_cdc_write_available()` that will make sure the internal FIFO is not overflow and still make the best use of it.